internal/config/configdomain/proposal_stack_lineage_builder_test.go

```diff
@@ -0,0 +1,173 @@
+
+func (self *mockConnectorProposalStackLineageBuilder) CreateProposal(_ forgedomain.CreateProposalArgs) error {
+	return nil
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) DefaultProposalMessage(_ forgedomain.ProposalData) string {
+	return "mock"
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) FindProposalFn() Option[func(branch, target gitdomain.LocalBranchName) (Option[forgedomain.Proposal], error)] {
+	return Some(func(branch, target gitdomain.LocalBranchName) (Option[forgedomain.Proposal], error) {
+		return Some(forgedomain.Proposal{
+			Data: forgedomain.ProposalData{
+				Body:         None[string](),
+				MergeWithAPI: false,
+				Number:       1,
+				Source:       branch,
+				Target:       target,
+				Title:        "Test Mocker",
+				URL:          "https://www.github.com/git-town/pull/1",
+			},
+			ForgeType: forgedomain.ForgeTypeCodeberg,
+		}), nil
+	})
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) OpenRepository(_ subshelldomain.Runner) error {
+	return nil
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) SearchProposalFn() Option[func(_ gitdomain.LocalBranchName) (Option[forgedomain.Proposal], error)] {
+	return None[func(_ gitdomain.LocalBranchName) (Option[forgedomain.Proposal], error)]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) SquashMergeProposalFn() Option[func(_ int, _ gitdomain.CommitMessage) error] {
+	return None[func(_ int, _ gitdomain.CommitMessage) error]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) UpdateProposalBodyFn() Option[func(_ forgedomain.ProposalInterface, _ string) error] {
+	return None[func(_ forgedomain.ProposalInterface, _ string) error]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) UpdateProposalSourceFn() Option[func(_ forgedomain.ProposalInterface, _ gitdomain.LocalBranchName) error] {
+	return None[func(_ forgedomain.ProposalInterface, _ gitdomain.LocalBranchName) error]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) UpdateProposalTargetFn() Option[func(_ forgedomain.ProposalInterface, _ gitdomain.LocalBranchName) error] {
+	return None[func(_ forgedomain.ProposalInterface, _ gitdomain.LocalBranchName) error]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) VerifyConnection() forgedomain.VerifyConnectionResult {
+	return forgedomain.VerifyConnectionResult{
+		AuthenticatedUser:   None[string](),
+		AuthenticationError: nil,
+		AuthorizationError:  nil,
+	}
+}
+
+func TestProposalStackLineageBuilder_CheckLineageAndProposals(t *testing.T) {
+	t.Parallel()
+	// arrange
+	mainBranch := gitdomain.NewLocalBranchName("main")
+	perennialBranch := gitdomain.NewLocalBranchName("development")
+	featureBranch := gitdomain.NewLocalBranchName("git-town/proposal-stack-lineage")
+	lineage := configdomain.NewLineage()
+	lineage.Root(mainBranch)
+	lineage.Set(perennialBranch, mainBranch)
+	lineage.Set(featureBranch, perennialBranch)
```

Comment:
Easier way to create a prepopulated lineage:

```suggestion
	lineage := configdomain.NewLineageWith(configdomain.LineageData{
		perennialBranch: mainBranch, 
		featureBranch: perennialBranch,
	})
```

Note that `perennialBranch: mainBranch` is wrong. Perennial branches have no parent branches.

Reply:
https://github.com/git-town/git-town/pull/4871/commits/aa09b6972caf3641efbce7654272488e54f6702a